transactions/base_task.go

```diff
@@ -46,22 +57,29 @@ func (b *BaseTask) UpdateTransaction(accountID identity.CentID, name string, err
 
 	if err != nil {
 		log.Infof("Transaction failed: %v\n", b.TxID.String())
-	} else {
-		log.Infof("Transaction successful:%v\n", b.TxID.String())
+		return errors.AppendError(err, b.updateStatus(accountID, Failed, NewLog(name, err.Error())))
 	}
 
-	tx, erri := b.TxService.GetTransaction(accountID, b.TxID)
-	if erri != nil {
-		return errors.AppendError(err, erri)
+	if b.Next {
+		err = b.updateStatus(accountID, Pending, NewLog(name, ""))
+		if err != nil {
```

Comment:
I think you can remove the IF. instead just return the updateStatus return.

Reply:
This was something we have discussed about having same transaction I'd for multiple tasks. Else we will not be able to communicate the next yxid to the client